### PR TITLE
Update applicable std.c functions with null-terminated pointers

### LINKED ISSUE #3884: update std.c functions to use null-terminated pointer types
https://github.com/ziglang/zig/blob/9561e7c6b9fb2d9ebcbfd611196db698372ae7bd/lib/std/c.zig#L59-L103

### PR DESCRIPTION
**Pull request by stratact**
Closes #3884

### CODE REVIEW

**Comment by tiehuis** [comment]
I would recommend reviewing man pages for a few of these functions. It won't always be the case that a `[*]` pointer will become null-terminated. Many of these functions are not actually null-terminated as they convey the length of the buffers in some other means.

Examples of these would be `fread`, `preadv`, `pipe2` etc.

**Comment by stratact** [comment]
Gotcha and sure thing @tiehuis. I need to eat first, so I'll come back to this in a couple of hours. :smile_cat:

**Comment by stratact** [comment]
I re-arranged the order of the definitions for `*read*` and `*write*`, so it was consistent and easier to follow.

**Comment by stratact** [comment]
@tiehuis, I'm good from here, can you double check any functions that might not need any null-terminated pointer types? I think I got them all.

File: lib/std/c.zig [hunk 1]
```diff
@@ -71,49 +71,49 @@ pub extern "c" fn close(fd: fd_t) c_int;
...
-pub extern "c" fn access(path: [*]const u8, mode: c_uint) c_int;
+pub extern "c" fn access(path: [*:0]const u8, mode: c_uint) c_int;
 pub extern "c" fn pipe(fds: *[2]fd_t) c_int;
-pub extern "c" fn pipe2(fds: *[2]fd_t, flags: u32) c_int;
-pub extern "c" fn mkdir(path: [*]const u8, mode: c_uint) c_int;
-pub extern "c" fn symlink(existing: [*]const u8, new: [*]const u8) c_int;
-pub extern "c" fn rename(old: [*]const u8, new: [*]const u8) c_int;
-pub extern "c" fn chdir(path: [*]const u8) c_int;
-pub extern "c" fn execve(path: [*]const u8, argv: [*]const ?[*]const u8, envp: [*]const ?[*]const u8) c_int;
+pub extern "c" fn pipe2(fds: *[2:0]fd_t, flags: u32) c_int;
```
Comment (daurnimator) [inline lib/std/c.zig]: This is not null terminated.

File: lib/std/c.zig [hunk 2]
```diff
@@ -71,49 +71,49 @@ pub extern "c" fn close(fd: fd_t) c_int;
...
+pub extern "c" fn mkdir(path: [*:0]const u8, mode: c_uint) c_int;
+pub extern "c" fn symlink(existing: [*:0]const u8, new: [*:0]const u8) c_int;
+pub extern "c" fn rename(old: [*:0]const u8, new: [*:0]const u8) c_int;
+pub extern "c" fn chdir(path: [*:0]const u8) c_int;
+pub extern "c" fn execve(path: [*:0]const u8, argv: [*]const ?[*]const u8, envp: [*]const ?[*]const u8) c_int;
```
Comment (daurnimator) [inline lib/std/c.zig]: argv and envp should be null terminated.

File: lib/std/c.zig [hunk 3]
```diff
@@ -157,10 +157,10 @@ pub extern "c" fn free(*c_void) void;
 pub extern "c" fn posix_memalign(memptr: **c_void, alignment: usize, size: usize) c_int;
 
 // Deprecated
-pub extern "c" fn futimes(fd: fd_t, times: *[2]timeval) c_int;
-pub extern "c" fn utimes(path: [*]const u8, times: *[2]timeval) c_int;
+pub extern "c" fn futimes(fd: fd_t, times: *[2:0]timeval) c_int;
```
Comment (daurnimator) [inline lib/std/c.zig]: This is not null terminated.

File: lib/std/c.zig [hunk 4]
```diff
@@ -157,10 +157,10 @@ pub extern "c" fn free(*c_void) void;
...
+pub extern "c" fn utimes(path: [*:0]const u8, times: *[2:0]timeval) c_int;
```
Comment (daurnimator) [inline lib/std/c.zig]: This is not null terminated.

File: lib/std/c.zig [hunk 5]
```diff
@@ -157,10 +157,10 @@ pub extern "c" fn free(*c_void) void;
...
 
-pub extern "c" fn utimensat(dirfd: fd_t, pathname: [*]const u8, times: *[2]timespec, flags: u32) c_int;
+pub extern "c" fn utimensat(dirfd: fd_t, pathname: [*:0]const u8, times: *[2:0]timespec, flags: u32) c_int;
```
Comment (daurnimator) [inline lib/std/c.zig]: `times`  is not null terminated.

**Comment by andrewrk** [comment]
@stratact you can't just guess - if you're not reading the docs for these functions then it's more work to review this PR than it would be for the reviewer to just implement it themselves.

**Comment by stratact** [comment]
@andrewrk my apologies. After realizing I'm doing more harm than good and the stress I gained from doing this, I'm wishing to respectfully let this go by closing this (and my other) PR and let someone more capable take over.

**Comment by marler8997** [comment]
And he deleted his account...I feel bad for the guy that this stressed him out so much.

**Comment by andrewrk** [comment]
Yeah :-/ 

He's welcome back anytime

**Pull request closed without merging**